sympy/solvers/ode/single.py

```diff
@@ -665,3 +667,87 @@ def _get_general_solution(self, *, simplify: bool = True):
             ) * exp(-(1 - n)*Integral(P, x)))**(1/(1 - n))
         )
         return [gensol]
+
+
+class Factorable(SingleODESolver):
+
+    hint = "factorable"
+    has_integral = False
+
+    def _matches(self):
+        eq = self.ode_problem.eq
+        func = self.ode_problem.func.func
+        x = self.ode_problem.sym
+
+        def _ode_factorable_match(eq, func):
+            from sympy.polys.polytools import factor
+            eqs = factor(eq)
+            eqs = fraction(eqs)[0] # p/q =0, So we need to solve only p=0
+            eqns = []
+            r = None
+            if isinstance(eqs, Pow):
+                # if f(x)**p=0 then f(x)=0 (p>0)
+                if eqs.exp.is_positive:
+                    eq = eqs.base
+                if isinstance(eq, Pow):
+                    return None
+                else:
+                    r = _ode_factorable_match(eq, func)
+                    if r is None:
+                        r = {'eqns' : [eq]}
+                    return r
+
+            if isinstance(eqs, Mul):
+                fac = eqs.args
+                for i in fac:
+                    if i.has(func):
+                        eqns.append(i)
+                if len(eqns)>0:
+                    r = {'eqns' : eqns}
+            return r
+        self.check = _ode_factorable_match(eq,func(x))
+
+        if self.check:
+            return True
+        else:
+            return False
+    def _get_general_solution(self, *, simplify: bool = True):
+        func = self.ode_problem.func.func
+        x = self.ode_problem.sym
+        r"""
+        Solves equations having a solvable factor.
```

Comment:
Don't put code before the docstring. The docstring won't work if it isn't the first thing in the function